Install dependencies for Vultr

when I run `/dapp vultr_list` I got  `Error: Cannot find module '@vultr/vultr-node'`  add code to install the dependency. (to be sure it will work add same in all the commands for Vultr, as user can run any command first). 

![image](https://user-images.githubusercontent.com/35557159/76077160-32e12280-5fc6-11ea-8e5c-40b0b919493a.png)

--- a/vultr/packages/vultr/vultr_list.js
+++ b/vultr/packages/vultr/vultr_list.js
@@ -1,6 +1,21 @@
 'use strict';
 
 /**
+ * A small function to install dependency.
+ * @param {[string]} pkgs - array of nim packages
+ */
+async function install(pkgs) {
+  pkgs = pkgs.join(' ');
+  return new Promise((resolve, reject) => {
+    const { exec } = require('child_process');
+    exec(`npm install ${pkgs}`, (err) => {
+      if (err) reject(err);
+      else resolve();
+    });
+  });
+}
+
+/**
  * A small function that converts slack elements `context` and `section` to mattermost compatible markdown.
  * @param {object} element - Slack element
  * @param {string} client - name of the client
@@ -45,7 +60,14 @@ async function _command(params, commandText, secrets = {}) {
   const result = [];
 
   try {
-    const Vultr = require('@vultr/vultr-node');
+    let Vultr;
+    // to isntall @vultr/vultr-node dependency
+    try {
+      Vultr = require('@vultr/vultr-node');
+    } catch {
+      await install(['@vultr/vultr-node']);
+      Vultr = require('@vultr/vultr-node');
+    }   
 
     const {server} = Vultr.initialize({apiKey: vultrApiKey});
 

--- a/vultr/packages/vultr/vultr_reboot.js
+++ b/vultr/packages/vultr/vultr_reboot.js
@@ -1,4 +1,18 @@
 'use strict';
+/**
+ * A small function to install dependency.
+ * @param {[string]} pkgs - array of nim packages
+ */
+async function install(pkgs) {
+  pkgs = pkgs.join(' ');
+  return new Promise((resolve, reject) => {
+    const { exec } = require('child_process');
+    exec(`npm install ${pkgs}`, (err) => {
+      if (err) reject(err);
+      else resolve();
+    });
+  });
+}
 
 /**
  * A small function that converts slack elements `context` and `section` to mattermost compatible markdown.
@@ -45,7 +59,14 @@ async function _command(params, commandText, secrets = {}) {
   const result = [];
 
   try {
-    const Vultr = require('@vultr/vultr-node');
+    let Vultr;
+    // to isntall @vultr/vultr-node dependency
+    try {
+      Vultr = require('@vultr/vultr-node');
+    } catch {
+      await install(['@vultr/vultr-node']);
+      Vultr = require('@vultr/vultr-node');
+    } 
 
     const {server} = Vultr.initialize({apiKey: vultrApiKey});
 

--- a/vultr/packages/vultr/vultr_snap.js
+++ b/vultr/packages/vultr/vultr_snap.js
@@ -1,6 +1,21 @@
 'use strict';
 
 /**
+ * A small function to install dependency.
+ * @param {[string]} pkgs - array of nim packages
+ */
+async function install(pkgs) {
+  pkgs = pkgs.join(' ');
+  return new Promise((resolve, reject) => {
+    const { exec } = require('child_process');
+    exec(`npm install ${pkgs}`, (err) => {
+      if (err) reject(err);
+      else resolve();
+    });
+  });
+}
+
+/**
  * A small function that converts slack elements `context` and `section` to mattermost compatible markdown.
  * @param {object} element - Slack element
  * @param {string} client - name of the client
@@ -45,8 +60,15 @@ async function _command(params, commandText, secrets = {}) {
   const result = [];
 
   try {
-    const Vultr = require('@vultr/vultr-node');
-
+    let Vultr;
+    // to isntall @vultr/vultr-node dependency
+    try {
+      Vultr = require('@vultr/vultr-node');
+    } catch {
+      await install(['@vultr/vultr-node']);
+      Vultr = require('@vultr/vultr-node');
+    }
+    
     const {snapshot} = Vultr.initialize({apiKey: vultrApiKey});
 
     const {SNAPSHOTID} = await snapshot.create({SUBID: Number(subid)});

